.circleci/config.yml

```
@@ -163,18 +173,23 @@ workflows:
   test:
     jobs:
       - test-linux:
+          name: test-linux-<< matrix.python-version >>-dimod<< matrix.dimod-version >>
           matrix:
             parameters:
-              image:
-                - 3.9-buster
-                - 3.8-buster
-                - 3.7-stretch
-                - 3.6-jessie
-                - 3.5-jessie
+              python-version: &python-versions [3.6.8, 3.7.9, 3.8.9, 3.9.4]
+
+              # only ever put one version in here, it's a bit of a silly
+              # way to do it but it keeps this number near the other python
+              # versions
+              integration-test-python-version: &integration-python-versions [3.9.4]
```

You could promote `integration-test-python-version` to pipeline parameter, but I guess this works for now.

At the very least, though, I would change it to:
```suggestion
              python-version: &python-versions [3.6.8, 3.7.9, 3.8.9, &latest-python 3.9.4]
              integration-test-python-version: [*latest-python]
```

---

I rather like your "very least" solution :smile:

---

I think you'll need to update it in the macos matrix as well.

---

Oh, I see you already did that.